worker/src/main/scala/org/apache/celeborn/service/deploy/worker/FetchHandler.scala

```diff
@@ -214,49 +214,63 @@ class FetchHandler(
             s"$endIndex get file name $fileName from client channel " +
             s"${NettyUtils.getRemoteAddress(client.getChannel)}")
 
-          val streamId = chunkStreamManager.nextStreamId()
-          // we must get sorted fileInfo for the following cases.
-          // 1. when the current request is a non-range openStream, but the original unsorted file
-          //    has been deleted by another range's openStream request.
-          // 2. when the current request is a range openStream request.
-          if ((endIndex != Int.MaxValue) || (endIndex == Int.MaxValue && !fileInfo.addStream(
-              streamId))) {
+          var nonRangeUnsorted = false
+          // streamId is -1 means shuffle client does not have any behavior to perform with worker, including fetching chunk, closing stream etc.
+          var streamId = -1L
+          // We must get sorted fileInfo for the following cases:
+          // 1. The current request is a range openStream request.
+          // 2. The current request is a non-range openStream, but original unsorted file has been deleted by another range's openStream request.
```

Comment:
IMO, addStream cannot be changed as this way, probably original files would be deleted before addStream at line 232.